docbrown/db/src/algorithms/reciprocity.rs

```diff
@@ -0,0 +1,102 @@
+use itertools::Itertools;
+use crate::graph_window::{WindowedGraph, WindowedVertex};
+
+
+fn get_unique_out_neighbours(v: &WindowedVertex) -> Vec<u64> {
+    let mut all_out_neighbours: Vec<u64> = v.out_neighbours_ids().unique().collect();
```

Comment:
out_neighbours_ids() should return unique itself, if it doesn't this is a bug and we should fix it.